community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/sampling/IndexSamplingController.java

```diff
@@ -46,6 +47,8 @@
     private final boolean backgroundSampling;
     private final Lock samplingLock = new ReentrantLock( true );
 
+    private JobHandle backgroundSamplingHandle = JobHandle.NO_OP;
```

Comment:
I don't think there is much value in having a `JobHandle.NO_OP`. Can't we simply check for not null on stop?